core/src/main/java/org/mule/runtime/core/internal/registry/SimpleRegistry.java

```diff
@@ -140,21 +141,35 @@ private void injectFieldDependencies() throws InitialisationException {
       Class<?> dependencyType = field.getType();
 
       boolean nullToOptional = false;
-      if (dependencyType.equals(Optional.class)) {
+      boolean collection = false;
+      if (dependencyType.equals(Optional.class) || Collection.class.isAssignableFrom(dependencyType)) {
```

Comment:
Is this for the case in which it is an Optional and it is also assignableFrom(dependencyType) to manage precedence? Try to make it clearer if this is the case and refactor if this is not a case with two conditions to reduce cyclomatic complexity.